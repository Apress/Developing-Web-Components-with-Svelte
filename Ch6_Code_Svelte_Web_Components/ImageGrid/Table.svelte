<svelte:options tag="cobalt-table" />

<script>
  import Grid from "./Grid.svelte";
  import Cell from "./Cell.svelte";

  export let columnCount = "4";
  export let rowCount = "4";
  export let itemCount = "";
  export let border = "";
  export let placeholderImages = "false";
</script>

<Grid columns={columnCount} rows={rowCount} {border}>
  {#each { length: parseInt(itemCount, 0) } as _, i}
    <Cell {placeholderImages} />
  {/each}
</Grid>
