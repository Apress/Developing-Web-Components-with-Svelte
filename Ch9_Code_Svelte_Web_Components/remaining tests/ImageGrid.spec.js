import { render, screen } from '@testing-library/svelte';
import Table from "../src/lib/ImageGrid/Table.svelte";

describe("Tests for ImageGrid", () => {
  const props = { 
    columnCount: "1",
    rowCount: "4", 
    border: "none",
    placeholderImages: "false",
    itemCount: "12",
  };

  it("should render properly without images", async() => {
    const result = render(Table, { props });
    expect(() => result).not.toThrow();
  });

  it("should render properly if images added", async() => {
    const result = render(Table, { ...props, placeholderImages: "true" });
    expect(() => result).not.toThrow();
  });

  it("should render properly if images added with text labels", async() => {
    const result = render(Table, { ...props, placeholderImages: "true" });
    const imageCount = result.getAllByTestId("placeholderImage");
    const labelCount = result.getAllByTestId("placeholderLabel");
    expect(imageCount.length).toBe(12);
    expect(labelCount.length).toBe(12)
  }); 

  it("get a snapshot of component", () => {
    const tree = render(Table, props);
    expect(tree).toMatchSnapshot();
  })
});