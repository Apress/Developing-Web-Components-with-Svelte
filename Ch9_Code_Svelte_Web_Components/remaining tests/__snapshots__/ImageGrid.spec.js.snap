// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ImageGrid get a snapshot of component 1`] = `
{
  "component": Table {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        "1",
        "4",
        "12",
        "none",
        "false",
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {
        "border": 3,
        "columnCount": 0,
        "itemCount": 2,
        "placeholderImages": 4,
        "rowCount": 1,
      },
      "root": <div>
        <div
          class="svelte-1qstdq7"
          style="grid-template-rows: repeat(4, 1fr); grid-template-columns: repeat(1, auto);"
        >
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          <div
            class="cell svelte-1d0j6iu"
          >
            <div
              class="cellSize svelte-1d0j6iu"
            />
          </div>
          
        </div>
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <div
        class="svelte-1qstdq7"
        style="grid-template-rows: repeat(4, 1fr); grid-template-columns: repeat(1, auto);"
      >
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        <div
          class="cell svelte-1d0j6iu"
        >
          <div
            class="cellSize svelte-1d0j6iu"
          />
        </div>
        
      </div>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
